src/transaction/transaction.rs

```diff
@@ -542,62 +597,95 @@ struct TwoPhaseCommitter {
     start_version: u64,
     bg_worker: ThreadPool,
     rpc: Arc<PdRpcClient>,
-    for_update_ts: u64,
+    style: TransactionStyle,
     #[new(default)]
     undetermined: bool,
 }
 
 impl TwoPhaseCommitter {
-    async fn commit(mut self) -> Result<()> {
+    async fn commit(mut self) -> Result<u64> {
         if self.mutations.is_empty() {
-            return Ok(());
+            return Ok(0);
         }
-        self.prewrite().await?;
-        match self.commit_primary().await {
-            Ok(commit_version) => {
-                self.bg_worker
-                    .clone()
-                    .spawn_ok(self.commit_secondary(commit_version).map(|res| {
-                        if let Err(e) = res {
-                            warn!("Failed to commit secondary keys: {}", e);
-                        }
-                    }));
-                Ok(())
-            }
-            Err(e) => {
-                if self.undetermined {
-                    Err(Error::undetermined_error(e))
-                } else {
-                    Err(e)
+
+        let min_commit_ts = self.prewrite().await?;
+
+        if self.style.try_one_pc {
+            return Ok(min_commit_ts);
```

Comment:
Can we just return here? I think `try_one_pc == true` doesn't mean we really use 1PC.

Reply:
I should have set it to false if it wasn't used.

Reply:
May be better to rename it to one_pc after it's determined to use one pc or not.